scripts/globals/magic_utils/spell_damage.lua

```diff
@@ -0,0 +1,908 @@
+        end
+        if math.random() < 0.33 or caster:getMod(elementalObi[spellElement]) >= 1 or isHelixSpell == true then
+            dayAndWeather = dayAndWeather + 0.10
+        end
+    elseif weather == xi.magic.singleWeatherWeak[spellElement] then
+        if math.random() < 0.33 or caster:getMod(elementalObi[spellElement]) >= 1 or isHelixSpell == true then
+            dayAndWeather = dayAndWeather - 0.10
+        end
+    elseif weather == xi.magic.doubleWeatherStrong[spellElement] then
+        if caster:getMod(xi.mod.IRIDESCENCE) >= 1 then
+            if math.random() < 0.33 or caster:getMod(elementalObi[spellElement]) >= 1 or isHelixSpell == true then
+                dayAndWeather = dayAndWeather + 0.10
+            end
+        end
+        if math.random() < 0.33 or caster:getMod(elementalObi[spellElement]) >= 1 or isHelixSpell == true then
+            dayAndWeather = dayAndWeather + 0.25
+        end
+    elseif weather == xi.magic.doubleWeatherWeak[spellElement] then
+        if math.random() < 0.33 or caster:getMod(elementalObi[spellElement]) >= 1 or isHelixSpell == true then
+            dayAndWeather = dayAndWeather - 0.25
+        end
+    end
+
+    -- Calculate day bonus
+    if dayElement == spellElement then
+        dayAndWeather = dayAndWeather + caster:getMod(xi.mod.DAY_NUKE_BONUS) / 100 -- sorc. tonban(+1)/zodiac ring
+        if math.random() < 0.33 or caster:getMod(elementalObi[spellElement]) >= 1 or isHelixSpell == true then
+            dayAndWeather = dayAndWeather + 0.10
+        end
+    elseif dayElement == xi.magic.elementDescendant[spellElement] then
+        if math.random() < 0.33 or caster:getMod(elementalObi[spellElement]) >= 1 or isHelixSpell == true then
+            dayAndWeather = dayAndWeather - 0.10
+        end
+    end
+
+    -- Cap bonuses from both day and weather
+    if dayAndWeather > 1.4 then
+        dayAndWeather = 1.4
+    end
+
+    return dayAndWeather
+end
+
+function calculateMagicBonusDiff(caster, target, spell, spellId, skillType, spellElement)
+    local magicBonusDiff = 1 -- The variable we want to calculate
+    local casterJob      = caster:getMainJob()
+    local mab            = caster:getMod(xi.mod.MATT)
+    local mabCrit        = caster:getMod(xi.mod.MAGIC_CRITHITRATE)
+    local mDefBarBonus   = 0
+
+    -- Drain/Aspir (II) Exception
+    -- if spellId >= 245 and spellId <= 248 then
+    --     magicBonusDiff = 1 + caster:getMod(xi.mod.ENH_DRAIN_ASPIR) / 100
+
+    --     if spellId == 247 or spellId == 248 then
+    --         magicBonusDiff = magicBonusDiff + caster:getMerit(xi.merit.ASPIR_ABSORPTION_AMOUNT) / 100
+    --     end
+
+    --     return magicBonusDiff
+    -- end
+
+    -- Ninja spell bonuses
+    if skillType == xi.skill.NINJUTSU then
+        -- Ninja Category 2 merits.
+        mab = mab + caster:getMerit(xi.merit.NIN_MAGIC_BONUS)
+        -- Ninja Category 1 merits
+        if spellId >= 320 and spellId <= 322 then     -- Katon series.
```

Comment:
You should be able to use `xi.magic.spellFamily` for these checks:
https://github.com/LandSandBoat/server/blob/base/scripts/globals/spell_data.lua#L113

For checks (like ancient magic) I wouldn't be against helpers living somewhere so you can just call `xi.spell.isAncientMagic(spellId)`. Even if you've commented it well

Reply:
Few things:
**Elemental Seal:**
From personal testing(No proof, sorry), elemental seal seems to make your resist go up a tier. So if you have 50% SDT, it goes to from 1/2 to 1. If you have sub 50% SDT, it goes from 1/4 to 1/2. 

**Resist stats for players only**
I have testing for this and proof!
https://www.ffxiah.com/forum/topic/46016/first-and-final-line-of-defense-v20/107/#3570888

https://www.ffxiah.com/forum/topic/46016/first-and-final-line-of-defense-v20/109/#3571304

```
Enfeebling test on a 2 resist state enfeeble, via having level ~70 Colibris reflect Dispel. I had Saboteur Addle II on the mob and 4/5 Malignance, so MEVA should be capped.

Neutral dark resistance: 1 landed, 9 resisted
-3 dark resistance: 10 landed
```

So, basically..

If you have any negative resists to the element hitting you, it will always land for 1/2 resist minimum(including enfeebles)
If you do not have any positive resist to the element hitting you, you cannot resist to 1/8.